File: python/secret-handshake/secret_handshake.py
def commands(binary_str):
    handshake = [action for index, action in enumerate(['wink', 'double blink', 'close your eyes', 'jump']) if binary_str[4 - index] == '1']
    if binary_str[0] == '1':
        handshake.reverse()
    return handshake

File: cpp/secret-handshake/secret_handshake.cpp
#include "secret_handshake.h"
#include <string>
#include <vector>
#include <algorithm>

namespace secret_handshake {
    std::vector<std::string> commands(int num) {
        std::vector<std::string> output {};
        if (num & 1) output.emplace_back("wink");
        if (num & 2) output.emplace_back("double blink");
        if (num & 4) output.emplace_back("close your eyes");
        if (num & 8) output.emplace_back("jump");
        if (num & 16) std::reverse(output.begin(), output.end());
        return output;
    }
}  // namespace secret_handshake
